### PR TITLE
Adding DF upsample

### LINKED ISSUE #162: Add `DataFrame.upsample`
#### Describe your feature request

I noticed that `nodejs-polars` doesn't have bindings for `upsample`, which is unfortunate, as I'm working on a project that requires it. I would appreciate it if it could be added.

### PR DESCRIPTION
**Pull request by Bidek56**
Adding DF upsample to close #162

### CODE REVIEW

File: polars/dataframe.ts [hunk 1]
```diff
@@ -1966,6 +2052,27 @@ export const _DataFrame = (_df: any): DataFrame => {
...
+      } else {
+        timeColumn = opts.timeColumn;
+        by = opts.by;
+        offset = opts.offset;
+        every = opts.every;
+        maintainOrder = opts.maintainOrder ?? false;
+      }
+
+      by = by ?? new Array();
+      if (typeof by === "string") by = [by];
```
Comment (universalmind303) [inline polars/dataframe.ts:2068]:
```suggestion
      if (typeof by === "string") {
        by = [by];
      } else {
        by = by ?? []
      }
```

**Review by universalmind303 — COMMENTED** [review]
just 1 small suggestion. Overall looks good.

**Review by universalmind303 — APPROVED** [review]

File: polars/dataframe.ts [hunk 2]
```diff
@@ -283,13 +282,13 @@ export interface DataFrame
...
-   * >  df = pl.DataFrame({
+   * >  const df = pl.DataFrame({
    * ...    "foo": [1, 2, 3],
    * ...    "bar": [6.0, 7.0, 8.0],
    * ...    "ham": ['a', 'b', 'c'],
    * ...    "apple": ['a', 'b', 'c']
-   * ...  })
-   * >  df.drop(['ham', 'apple'])
+   * ...  });
+   * >  console.log(df.drop(['ham', 'apple']).toString());
```
Comment (universalmind303) [inline polars/dataframe.ts:291]: you shouldn't need `toString()` here. the `node.util.inspect` already properly displays the dataframe in `console.log`

Comment (Bidek56) [reply polars/dataframe.ts:291]: but It does not work in `Bun` b/c I get `ProxyObject` without `.toString()`:
```
ProxyObject {
  foo: shape: (3,)
Series: 'foo' [f64]
[
	1.0
	2.0
	3.0
],
  bar: shape: (3,)
Series: 'bar' [f64]
[
	6.0
	7.0
	8.0
],
}
```

Comment (universalmind303) [reply polars/dataframe.ts:291]: if that's the case, that'd be a bug in bun. The toString should be unnecessary.

Comment (Bidek56) [reply polars/dataframe.ts:291]: I [pinged](https://github.com/oven-sh/bun/issues/4121) Jarred about this issue. Thx

Comment (Bidek56) [reply polars/dataframe.ts:291]: I am not sure if it's a bun issues, b/c this works fine in bun:
```
console.log(
   pl.DataFrame([{ a: 1, b: 2,}])[Symbol.for("nodejs.util.inspect.custom")]()
);
```

It results in:
```
shape: (1, 2)
┌─────┬─────┐
│ a   ┆ b   │
│ --- ┆ --- │
│ f64 ┆ f64 │
╞═════╪═════╡
│ 1.0 ┆ 2.0 │
└─────┴─────┘
```

Comment (universalmind303) [reply polars/dataframe.ts:291]: yeah this is definitely an issue with bun. If they claim to be 100% interopable with node.js, then the outputs should be the same. I'd suggest opening an issue with them.

Comment (universalmind303) [reply polars/dataframe.ts:291]: yes the symbol works as expected, but console.logging without it does not. 

```js
console.log(pl.DataFrame({a: [1]}))
```

this should be identical in both. If it's not, then that is a bug in bun.

Comment (Bidek56) [reply polars/dataframe.ts:291]: [Checking](https://github.com/oven-sh/bun/pull/4243#issuecomment-1949106097) with the Bun team.

**Pull request merged**